components/embedder_traits/resources.rs

```diff
@@ -3,13 +3,19 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 use std::path::PathBuf;
-use std::sync::{Once, RwLock};
+use std::sync::RwLock;
 
 use lazy_static::lazy_static;
 
 lazy_static! {
-    static ref RES: RwLock<Option<Box<dyn ResourceReaderMethods + Sync + Send>>> =
-        RwLock::new(None);
+    static ref RES: RwLock<Option<Box<dyn ResourceReaderMethods + Sync + Send>>> = {
+        if cfg!(servo_production) {
+            RwLock::new(None)
+        } else {
+            const _: () = assert!(cfg!(servo_do_not_use_in_production));
```

Comment:
Do we need the assignment to `_` here to avoid some compiler warning?

Reply:
this is a [static assertion](https://doc.rust-lang.org/beta/reference/items/constant-items.html#evaluation); by calling assert!() in a const item, the assertion is run at compile time only

Reply:
Ah! I think this is causing [issue](https://github.com/servo/servo/pull/30509/files?diff=unified&w=0#r1372606678) when building with `./mach build --profile production`. Am I doing something wrong?

Reply:
no, it was broken for me too :sweat_smile:

Reply:
added comments to clarify these in f6bb63570ddbf208d5f209421a63db1b1a5af9cb